### PR TITLE
[GEOT-5632] Default PointPlacement anchorPoint does not match SLD 1.0

### PR DESCRIPTION
**Pull request by boundlessgeo-yancy**
The default point is now X=0, Y=0.5.

See https://osgeo-org.atlassian.net/browse/GEOT-5632

### CODE REVIEW

File: modules/library/api/src/test/java/org/geotools/styling/AnchorPointTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+public class AnchorPointTest {
+
+  @Test
+  public void testDefaultAnchorPoint() {
+    AnchorPoint anchorPoint = AnchorPoint.DEFAULT;
+
+    assertEquals(0.0, anchorPoint.getAnchorPointX().evaluate(null, Double.class), 0.0);
+    assertEquals(0.5, anchorPoint.getAnchorPointY().evaluate(null, Double.class), 0.0);
+  }
```
Comment (jodygarnett) [inline modules/library/api/src/test/java/org/geotools/styling/AnchorPointTest.java]: Test looks good, hold I thought only one of the values was 0.0.

From https://osgeo-org.atlassian.net/browse/GEOT-5632

> According to SLD v1.0 standard on page 47,
> The default point is X=0, Y=0.5, which is at the middle height of the lefthand
side of the label.

File: modules/library/render/src/main/java/org/geotools/renderer/style/SLDStyleFactory.java [hunk 2]
```diff
@@ -578,12 +578,12 @@ Style2D createPointStyle(Object feature, Symbolizer symbolizer, Graphic sldGraph
             displacementY = evalToFloat(displacement
 					.getDisplacementY(), feature, 0f);
 		}
-        float anchorPointX = 0.5f;
-        float anchorPointY = 0.5f;
+        float anchorPointX = AnchorPoint.DEFAULT.getAnchorPointX().evaluate(null, Float.class);
+        float anchorPointY = AnchorPoint.DEFAULT.getAnchorPointY().evaluate(null, Float.class);
```
Comment (jodygarnett) [inline modules/library/render/src/main/java/org/geotools/renderer/style/SLDStyleFactory.java]: Okay, making reuse of SLD defaults here.

File: modules/library/render/src/test/java/org/geotools/renderer/style/SLDStyleFactoryTest.java [hunk 3]
```diff
@@ -111,6 +111,10 @@ public void testCreatePointStyle(){
         NumberRange<Double> scaleRange = NumberRange.create( 1.0,50000.0);
         Style2D style = sld.createPointStyle(null, pointsym, scaleRange);
         assertNotNull( style );
+
+        PointStyle2D ps = (PointStyle2D) style;
+        assertEquals(0.0f, ps.getAnchorPointX(), 0.0f);
+        assertEquals(0.5f, ps.getAnchorPointY(), 0.0f);
```
Comment (jodygarnett) [inline modules/library/render/src/test/java/org/geotools/renderer/style/SLDStyleFactoryTest.java]: okay so you already have this covered, no need for AnchorPointTest above?

File: modules/library/api/src/test/java/org/geotools/styling/AnchorPointTest.java (hunk 1 above)
Comment (boundlessgeo-yancy) [inline modules/library/api/src/test/java/org/geotools/styling/AnchorPointTest.java]: The expected values are 0.0 for x and 0.5 for y. The 0.0 at the end of those lines are the delta values for float comparisons.

File: modules/library/render/src/test/java/org/geotools/renderer/style/SLDStyleFactoryTest.java (hunk 3 above)
Comment (boundlessgeo-yancy) [inline modules/library/render/src/test/java/org/geotools/renderer/style/SLDStyleFactoryTest.java]: I suppose so. That is fine with me to remove that other test.

File: modules/library/api/src/test/java/org/geotools/styling/AnchorPointTest.java (hunk 1 above)
Comment (jodygarnett) [inline modules/library/api/src/test/java/org/geotools/styling/AnchorPointTest.java]: sorry read the code wrong

**Comment by jodygarnett** [comment]
Okay so we have the expected test failures:

````
...
You can add -Dorg.geotools.image.test.interactive=true to show a dialog comparing them (requires GUI support)
  VectorFillTest.testFontFill:160->runSingleLayerTest:81->runSingleLayerTest:87->runSingleLayerTest:105 Images are visibly different, found 13898 different pixels, against a threshold of 100
You can add -Dorg.geotools.image.test.interactive=true to show a dialog comparing them (requires GUI support)
  VectorFillTest.testCrossFill:111->runSingleLayerTest:87->runSingleLayerTest:105 Images are visibly different, found 30285 different pixels, against a threshold of 300
You can add -Dorg.geotools.image.test.interactive=true to show a dialog comparing them (requires GUI support)
  WrapPointSymbolizerTest.testWrapPointSymbolizer:66 expected:<java.awt.Color[r=255,g=0,b=0]> but was:<java.awt.Color[r=0,g=0,b=0]>

Tests run: 868, Failures: 143, Errors: 0, Skipped: 2
````

**Comment by jodygarnett** [comment]
Okay, doubling back to the bug report - we have a different default value between SLD 1.0 and SE 1.1 (sigh). So there is not going to be a clear right answer. 

I think we could go for the SLD 1.0 default; and double check that the SE 1.1 parser fills in the correct value when it creates the LabelPlacement.

**Comment by jodygarnett** [comment]
Thanks @boundlessgeo-yancy - adding "org.geotools.renderer.style.legacyAnchorPoint" takes some of the risk out of this change for those upgrading. You are a champ.

**Comment by jodygarnett** [comment]
merged, thanks for taking part in the bug stomp.

**Pull request closed without merging**